### DISCUSSION TITLE
While exporting, Auto disable plugin on error

### DISCUSSION THREAD

**Discussion by linonetwo** [TiddlyWiki/TiddlyWiki5#6861] [Ideas]
One user in our QQ group reported echarts plugin blocks exporting. (This happened on my side too)

![20208C5A7ABCA5BF97D570F7C969CCF4](https://user-images.githubusercontent.com/3746270/182025632-021b98c9-c907-410b-aea5-63b61e14c341.jpg)

From the error stack, we can see the `$:/plugins/Gk0wk/echarts`

![AF6ABA2854B180BBAA3002B33C3D8D0B](https://user-images.githubusercontent.com/3746270/182025665-5ca8c5c1-a1bc-4259-8a6a-5632a460e3c5.jpg)

So I guide this user to disabling the echarts plugin, continuing his exportion.

I think one solution to this can be Auto disable plugins that show up in the error stack, and Auto reenable them after export, what do you think?

**Comment by pmario** [comment]
Hi, 

It would be nice to have a "To reproduce" step by step info, so we are able to reproduce the problem ourselves. 

Otherwise my advice will be as follows: 

I think the right thing to do, would be to fix the problem in the plugin instead of creating a workaround in the core.

**Comment by Jermolene** [comment]
Hi @linonetwo 

> I think one solution to this can be Auto disable plugins that show up in the error stack, and Auto reenable them after export, what do you think?

The trouble is that we can't disable JS modules in plugins without a reload; there's no way to disable plugins once they've executed.

The usual escape hatch when it is not possible to save or export would be to use a bookmarklet to copy the JSON of all tiddlers to the clipboard, from where it can be saved in a file. It would be helpful perhaps to publish that bookmarklet on tiddlywiki.com

**Reply by linonetwo** [reply]
Oh, I forget this...

In this case, user just want the HTML export. Maybe I can refine that error message to guide user try disabling the plugin. (Instead of needing to open dev tool and ask in the group)

**Comment by pmario** [comment]
I did test to create a New Tiddler at: https://tiddly-gittly.github.io/tw-echarts/  ... When I do save the tiddler, the whole wiki is automatically saved as a file. ... So it seems something has been done with the auto-save function because that's not standard. It doesn't work that way at tiddlywiki.com

For me it seems the "click" event "tm-download-file" is broken. It seems the event.propagation isn't stopped.

**Reply by pmario** [reply]
I did try: 

```
title: New Tiddler

<<echarts "$:/plugins/Gk0Wk/echarts/addons/SimpleExample.json">>
```

Export Tiddler : Static HTML

is the echarts-macro able to export it's output as a static image?

**Reply by linonetwo** [reply]
Yes, maybe this is better get fixed from @Gk0Wk 's side. He have some effort to make exporting work. I don'k know why standard "tm-download-file" is not working now.

**Reply by pmario** [reply]
I did single step through the saving mechanism. It seemed to create the message and also activated the save actions, but then it seemed to go on with .dispatchEvent() ... It seems this event bubbles up until there is no parent node anymore. ... At least, that's what I think
